app/src/main/java/org/p2p/wallet/history/ui/historylist/HistoryListView.kt

```diff
@@ -1,17 +1,184 @@
 package org.p2p.wallet.history.ui.historylist
 
+import androidx.core.view.isVisible
+import androidx.lifecycle.Lifecycle
+import androidx.recyclerview.widget.LinearLayoutManager
 import android.content.Context
 import android.util.AttributeSet
 import android.widget.FrameLayout
+import com.google.android.material.snackbar.Snackbar
 import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.koin.core.parameter.parametersOf
+import org.p2p.core.common.TextContainer
+import org.p2p.core.glide.GlideManager
+import org.p2p.core.token.Token
+import org.p2p.uikit.utils.attachAdapter
 import org.p2p.uikit.utils.inflateViewBinding
+import org.p2p.uikit.utils.recycler.RoundedDecoration
+import org.p2p.wallet.common.ui.recycler.EndlessScrollListener
+import org.p2p.wallet.common.ui.recycler.PagingState
 import org.p2p.wallet.databinding.LayoutHistoryListBinding
+import org.p2p.wallet.history.model.HistoryItem
+import org.p2p.wallet.history.ui.token.adapter.HistoryAdapter
 
 class HistoryListView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
-) : FrameLayout(context, attrs, defStyleAttr), KoinComponent {
+) : FrameLayout(context, attrs, defStyleAttr), KoinComponent, HistoryListViewContract.View {
 
     private val binding = inflateViewBinding<LayoutHistoryListBinding>()
+
+    private var tokenForHistory: Token.Active? = null
+
+    private val glideManager: GlideManager by inject()
+
+    private lateinit var historyAdapter: HistoryAdapter
+
+    private val presenter: HistoryListViewContract.Presenter by inject { parametersOf(tokenForHistory) }
+
+    private var isBound = false
```

Comment:
if you know how to avoid using this 
please tell me
we need `by inject` to call in needed place and `onAttachedToWindow` should not call `presenter.attach(this)` before `bind` called

Reply:
https://developer.android.com/reference/kotlin/androidx/core/view/package-summary#(android.view.View).doOnAttach(kotlin.Function1)